Documentation/ceph-cluster-crd.md

```diff
@@ -878,22 +878,162 @@ So just taking the `storage` section this will give something like:
 ```
 
-> **NOTE**: Note that Rook only supports two naming convention for a given template:
+> **NOTE**: Note that Rook only supports three naming convention for a given template:
 
-* "data": represents the main OSD block device, where your data are being stored
-* "metadata": represents the metadata device used to store the Ceph Bluestore database for an OSD.
-It is recommended to use a faster storage class for the metadata device, with a slower device for the data.
+* "data": represents the main OSD block device, where your data is being stored.
+* "metadata": represents the metadata (including block.db and block.wal) device used to store the Ceph Bluestore database for an OSD.
+* "wal": represents the block.wal device used to store the Ceph Bluestore database for an OSD. If this device is set, "metadata" device will refer specifically to block.db device.
+It is recommended to use a faster storage class for the metadata or wal device, with a slower device for the data.
 Otherwise, having a separate metadata device will not improve the performance.
-To determine the size of the metadata block follow the [official Ceph sizing guide](https://docs.ceph.com/docs/mimic/rados/configuration/bluestore-config-ref/#sizing).
+
+The bluestore partition has the following reference conbinations, this supported by ceph-volume utility:
+
+* A single "data" device.
+```yaml
+  storage:
+   storageClassDeviceSets:
+    - name: set1
+      count: 3
+      portable: false
+      tuneDeviceClass: false
+      volumeClaimTemplates:
+      - metadata:
+          name: data
+        spec:
+          resources:
+            requests:
+              storage: 10Gi
+          # IMPORTANT: Change the storage class depending on your environment (e.g. local-storage, gp2)
+          storageClassName: gp2
+          volumeMode: Block
+          accessModes:
+            - ReadWriteOnce
+```
+
+* A "data" device and a "metadata" device.
+```yaml
+  storage:
+   storageClassDeviceSets:
+    - name: set1
+      count: 3
+      portable: false
+      tuneDeviceClass: false
+      volumeClaimTemplates:
+      - metadata:
+          name: data
+        spec:
+          resources:
+            requests:
+              storage: 10Gi
+          # IMPORTANT: Change the storage class depending on your environment (e.g. local-storage, gp2)
+          storageClassName: gp2
+          volumeMode: Block
+          accessModes:
+            - ReadWriteOnce
+      - metadata:
+          name: metadata
+        spec:
+          resources:
+            requests:
+              # Find the right size https://docs.ceph.com/docs/master/rados/configuration/bluestore-config-ref/#sizing
+              storage: 5Gi
+          # IMPORTANT: Change the storage class depending on your environment (e.g. local-storage, io1)
+          storageClassName: io1
+          volumeMode: Block
+          accessModes:
+            - ReadWriteOnce
+```
+
+* A "data" device and a "wal" device.
```

Comment:
Do we need to document this case? I thought if they want to specify a `wal` device, they should likely also specify a `metadata` device.

Reply:
Do you means add the description in Rook documentation in this case? not sure where to add.
I just add some description in this case of ceph-cluster-crd.md.